packages/atoms/connect/ConnectButton.tsx

```diff
@@ -0,0 +1,68 @@
+import { cn } from "@/lib/utils";
+import { useState } from "react";
+import * as React from "react";
+
+import { Button } from "../src/components/ui/button";
+import type { AtomsGlobalConfigProps } from "../types";
+
+// we accept two props: buttonText & onButtonClick
+// buttonText is for displaying buttonText
+// onButtonClick is a a method to handle onClick for button
+// it accepts another object which contains a callback in case of any error
```

Comment:
1. I don't see `onButtonClick` prop in `ConnectButtonProps`

2. For all atoms it's simpler to just always accept `onClick` handler instead of `onXClick`, but as Morgan suggested (great suggestion btw) we can just extend `React.ButtonHTMLAttributes<HTMLButtonElement>` which already includes `onClick` there. Here is the shadcn Button example: https://github.com/shadcn-ui/taxonomy/blob/main/components/ui/button.tsx

3. I would also remove all of the comments from line 8 - 11, because the prop names themselves should explain what is going on and comments in this case are redundant - they just restate what developers can see in code

Reply:
Sorry about the comments, I kinda wrote them just at the start of the component for my own thought process and forgot to update them as I made changes, will remove them altogether. Nice catch for the `onClick` prop, I actually forgot to remove that from the props.